services/mlmodel/server.go

```diff
@@ -28,24 +30,41 @@ func (server *serviceServer) Infer(ctx context.Context, req *pb.InferRequest) (*
 	if err != nil {
 		return nil, err
 	}
-	id, err := asMap(req.InputData)
+	var id map[string]interface{}
+	if req.InputData != nil {
+		id, err = asMap(req.InputData)
+		if err != nil {
+			return nil, err
+		}
+	}
+	var it ml.Tensors
+	if req.InputTensors != nil {
+		it, err = ProtoToTensors(req.InputTensors)
+		if err != nil {
+			return nil, err
+		}
+	}
+	ot, od, err := svc.Infer(ctx, it, id)
```

Comment:
I'd consider rejecting requests with both pb.Struct and Tensor inputs with an error.

Reply:
made the change